dali/operators/decoder/audio/generic_decoder.h

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_GENERIC_DECODER_H_
+#define DALI_OPERATORS_DECODER_AUDIO_GENERIC_DECODER_H_
+
+#include <sndfile.h>
+#include <cassert>
+#include <cstring>
+#include <memory>
+#include <vector>
+#include "dali/core/format.h"
+#include "dali/core/error_handling.h"
+#include "dali/operators/decoder/audio/audio_decoder.h"
+
+
+namespace dali {
+
+namespace detail {
+
+template<typename SampleType>
```

Comment:
I believe everything related to implementation (e.g. libsnd data structures) should be moved to a `*.cc` file. I think most of the code should, if not all

Reply:
Done